src/test/solution_space/solution_space_serializer/solution_space_serializer_test.py

```diff
@@ -46,3 +49,9 @@ def test_graph_representation(self, param_solution_space_serializer_test: Callab
         deserialized_graph = SolutionSpaceSerializer.deserialize(serialized_path)
         assert are_graph_folder_structures_equal(graph.graph_directory, deserialized_graph.graph_directory)
         assert graph.get_pretty_string() == deserialized_graph.get_pretty_string()
+
+    def test_id_counter_serialization(self) -> None:
```

Comment:
It is a strange method. You don't use assert or others. Maybe is it better to use set_item_by_id method here with assert id's for example?